internal/elasticsearch/index/indices/schema.go

```diff
@@ -0,0 +1,434 @@
+package indices
+
+import (
+	"context"
+	"regexp"
+
+	"github.com/elastic/terraform-provider-elasticstack/internal/elasticsearch/index"
+	"github.com/elastic/terraform-provider-elasticstack/internal/utils/customtypes"
+	"github.com/hashicorp/terraform-plugin-framework-jsontypes/jsontypes"
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+func (d *dataSource) Schema(_ context.Context, _ datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = getSchema()
+}
+
+func getSchema() schema.Schema {
+	return schema.Schema{
+		Description: "Manages Elasticsearch indices. See: https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-get-index.html",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description: "Generated ID for the indices.",
+				Computed:    true,
+			},
+			"search": schema.StringAttribute{
+				Description: "Comma-separated list of indices to resolve by their name. Supports wildcards `*`.",
+				Optional:    true,
+			},
```

Comment:
```suggestion
			"target": schema.StringAttribute{
				Description: "Comma-separated list of data streams, indices, and aliases used to limit the request. Supports wildcards (*). To target all data streams and indices, omit this attribute or use * or _all. ",
				Optional:    true,
			},
```

Copying the details from the ES [docs](https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-get-index.html#get-index-api-path-params). I'm not a huge fan of the `search` name, and would have thought of this as `name` or `name_pattern` or similar but in reality it could also be an alias resolving to multiple indices so sticking with `target` might make the most sense. WDYT?